### PR TITLE
Some create timer modal improvements (SPUR)

### PR DESCRIPTION
**Pull request by KKoukiou**
Partially addresses https://github.com/cockpit-project/cockpit/issues/15927

### CODE REVIEW

**Comment by garrett** [comment]
Dropdowns are center aligned... and I'm not sure why.

Screenshot from 2021-06-16 13-24-30

Hourly timers are still non-obvious. (I guess that's minute offset?)

image

- We're creating a single timer in the UI, so the dialog should be "Create timer", not "Create timers".
- Is there a limit or a recommended length on service names? If so, we should show that in the UI. 
- Are there only ever two options for "Run"? It should probably be renamed "When" as it's after system boot or at a specific time.
- After system boot is a delay, right? It should probably be called "Delay"?

Some sketches:

service-timer-modal(1)

service-timer-modal(2)

service-timer-modal(4)

**Comment by garrett** [comment]
I'm fine if this PR just adjusts the list above, including the simle stuff like alignment and the text strings. UI changes in this case are mainly superfluous and can go in a follow-up or just be ignored, as what we have isn't so bad (except for the lack of communication of what that text entry is for hourly mode and the "Run"/"When" dropdown for just two items).

**Comment by KKoukiou** [comment]
> I'm fine if this PR just adjusts the list above, including the simle stuff like alignment and the text strings. UI changes in this case are mainly superfluous and can go in a follow-up or just be ignored, as what we have isn't so bad (except for the lack of communication of what that text entry is for hourly mode and the "Run"/"When" dropdown for just two items).

I can fix your suggestions in this PR - it's about getting the perfect from this anyway. When we are satisfied I will also add a pixel test to make sure we don't break it.

**Comment by KKoukiou** [comment]
@garrett there are still some unsolved issue from https://github.com/cockpit-project/cockpit/issues/15927#issuecomment-862108330 which are not so trivial.

Can you please review this again anyway? and maybe we can split the enhancements in multiple PRs.

**Review by garrett — APPROVED** [review]
I like the changes and I'm happy with splitting off the rest to another PR.

There is a little problem with the layout in super-small sizes (iPhone SE), but everything else is fine.

Screen Shot 2021-06-17 at 16 48 01


One somewhat notable thing: When adding new items, they're below the scroll. We should probably instruct the browser to scroll down to the bottom when adding them. When there's no overflow, it shouldn't do anything. This mainly affects smaller screens like mobile.

But, overall, happy to see a lot of this custom code go away, and the rest can be addressed in follow-up PR(s).

**Review by garrett — CHANGES_REQUESTED** [review]
Oh, I just noticed the radio button labels aren't clickable. This means it's also broken for accessibility.

Screenshot from 2021-06-17 17-01-01

The label exists, but it doesn't have a for="" attribute pointing to the ID of the `<input>`... and the input also doesn't have a generated unique ID.

File: pkg/systemd/services/timer-dialog.jsx [hunk 1]
```diff
@@ -159,153 +164,189 @@ const CreateTimerDialogBody = ({ setIsOpen }) => {
...
+                        <Radio value="system-boot"
+                               name="boot-or-specific-time"
+                               onChange={() => setDelay("system-boot")}
+                               isChecked={delay == "system-boot"}
+                               label={_("After system boot")} />
+                        <Radio value="specific-time"
+                               name="boot-or-specific-time"
+                               onChange={() => setDelay("specific-time")}
+                               isChecked={delay == "specific-time"}
+                               label={_("At specific time")} />
```
Comment (garrett) [inline pkg/systemd/services/timer-dialog.jsx:178]: I guess leaving out an ID doesn't make the radios get hooked up properly in PatternFly, so each needs its own unique ID.

This really should be a compile time error for PF, or PF should wrap the label around the radio. Either-or, but right now, neither happens, so we need to manually ensure there's an ID.

**Review by garrett — DISMISSED** [review]
1. Dropdowns look fine (that was my fault, and on my machine only)
   Screenshot from 2021-06-21 16-09-06
2. Labels are now fixed for a11y and are clickable!

LGTM! :+1:

**Review by garrett — APPROVED** [review]

**Pull request merged**